### DISCUSSION TITLE
Bundle

### DISCUSSION THREAD

**Discussion by DasLixou** [bevyengine/bevy#6778] [Q&A]
Is there a way to make a trait with a function which returns a Bundle? When I do it with `-> dyn Bundle` it says it cannot be made into an object

**Comment by bnyu** [comment]
There is no way to use `dyn Bundle` since the `Bundle` trait is not [object-safety](https://doc.rust-lang.org/reference/items/traits.html#object-safety) at bevy 0.9
see #3227 tracking progress

For now you can do something like this as an alternative:
```
pub trait MyBundle {
    fn components() -> Vec<Box<dyn Component<Storage=TableStorage>>>;
}
```
